ambry-api/src/test/java/com.github.ambry/router/ByteRangeTest.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.router;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+
+public class ByteRangeTest {
```

Comment:
I hope you are testing off-by-one cases to ensure that both startOffset and endOffset get included and so on.